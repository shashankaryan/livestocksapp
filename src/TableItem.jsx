// @flow

import React, { Component } from 'react'

class TableItem extends Component {
	state = {}
	props: Item
	render() {
		return (
			<tr>
				<td>{this.props.name.toUpperCase()}</td>
				<td id={this.props.name}>{this.props.price}</td>
				<td>{this.props.updateDuration}</td>
			</tr>
		)
	}
}
export default TableItem
