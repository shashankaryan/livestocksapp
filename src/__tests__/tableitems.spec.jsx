import React from 'react'
import { shallow } from 'enzyme' // render, static is also available. but they are slower.
import TableItem from '../TableItem'

test('TableItem should render correctly based on item', () => {
	const item = { name: 'link', price: 127.9, updateDuration: 'A Few Seconds Ago...' }
	const component = shallow(<TableItem key={item.name} {...item} />)
	expect(component).toMatchSnapshot()
})
