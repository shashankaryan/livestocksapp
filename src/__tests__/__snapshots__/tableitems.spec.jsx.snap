// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableItem should render correctly based on item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr>
    <td>
        LINK
    </td>
    <td
        id="link"
    >
        127.9
    </td>
    <td>
        A Few Seconds Ago...
    </td>
</tr>,
  "nodes": Array [
    <tr>
      <td>
            LINK
      </td>
      <td
            id="link"
      >
            127.9
      </td>
      <td>
            A Few Seconds Ago...
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TableItem
        name="link"
        price={127.9}
        updateDuration="A Few Seconds Ago..."
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TableItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "name": "link",
          "price": 127.9,
          "updateDuration": "A Few Seconds Ago...",
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr>
          <td>
                    LINK
          </td>
          <td
                    id="link"
          >
                    127.9
          </td>
          <td>
                    A Few Seconds Ago...
          </td>
</tr>,
        "_debugID": 2,
        "_renderedOutput": <tr>
          <td>
                    LINK
          </td>
          <td
                    id="link"
          >
                    127.9
          </td>
          <td>
                    A Few Seconds Ago...
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TableItem
    name="link"
    price={127.9}
    updateDuration="A Few Seconds Ago..."
/>,
}
`;
